lib/motion-spec/mock/mock.rb

```diff
@@ -0,0 +1,65 @@
+# -*- encoding : utf-8 -*-
+class Object
+  # Create a mock method on an object.  A mock object will place an expectation
+  # on behavior and cause a test failure if it's not fulfilled.
+  #
+  # ==== Examples
+  #
+  #    my_string = "a wooden rabbit"
+  #    my_string.mock!(:retreat!, :return => "run away!  run away!")
+  #    my_string.mock!(:question, :return => "what is the airspeed velocity of an unladen sparrow?")
+  #
+  #    # test/your_test.rb
+  #    my_string.retreat!    # => "run away!  run away!"
+  #    # If we let the test case end at this point, it fails with:
+  #    # Unmet expectation: #<Sparrow:1ee7> expected question
+  #
+  def mock!(method, options = {}, &block)
+    MotionSpec::Mocks.add([self, method])
+
+    behavior =
+      if block_given?
+        lambda do |*args|
+          raise ArgumentError if block.arity >= 0 && args.length != block.arity
+
+          MotionSpec::Mocks.verify([self, method])
+          block.call(*args)
+        end
+      elsif !options[:yield].nil?
+        lambda do |*args|
+          MotionSpec::Mocks.verify([self, method])
+          yield(options[:yield])
+        end
+      else
+        lambda do |*args|
```

Comment:
Unused block argument - `args`. If it's necessary, use `_` or `_args` as an argument name to indicate that it won't be used. Also consider using a proc without arguments instead of a lambda if you want it to accept any arguments but don't care about them.